extension/lib/shopgate/CartExtensionPipeline.js

```diff
@@ -27,13 +27,31 @@ class ShopgateCartExtensionPipeline {
    * @returns {Promise<void>}
    */
   async addProducts (products) {
-    const bigCommerceLineItems = products.map((product) => {
+    const bigCommerceCart = await this._bigCommerceCartRepository.load()
+
+    const itemsToAdd = []
+    const itemsToUpdate = []
+
+    await Promise.all(products.map(async (product) => {
       const {productId, variantId} = this._identifierConverter.extractProductIds(product.productId)
 
-      return BigCommerceCartRepository.createLineItem(productId, product.quantity, variantId)
-    })
+      let found = null
+      if (bigCommerceCart && bigCommerceCart.lineItems) {
+        found = bigCommerceCart.lineItems._items.find(bcItem => bcItem._productId === productId)
+      }
+      if (found) {
+        itemsToUpdate.push(BigCommerceCartRepository.createLineItemUpdate(found._id, parseInt(product._quantity) + parseInt(found._quantity)))
+      } else {
```

Comment:
when I tested this I couldn't add another of the same item to the cart. I did some testing and I think there is a problem on line 43. I think product._quantity should be product.quantity. When I console out product at this point { productId: '123', quantity: 1 } is returned.

Reply:
you're right, i fixed that